### PR TITLE
Sparrow dom/origin wallet twitter workaround

### PR DESCRIPTION
**Pull request by sparrowDom**
### Description:
Twitter does not display the share dialog with the user agent that is default to react Native webView. For that reason we replace that user agent with a working one

**Solves issue:**https://github.com/OriginProtocol/origin/issues/2844

### Checklist:

- [x] Test your work and double-check to confirm that you didn't break anything
- [ ] [Wrap any new text/strings with `fbt`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#wrapping-text) for translation
- [ ] If there are any new/changed strings to translate, [`npm run translate`](https://github.com/OriginProtocol/origin/tree/master/dapps/marketplace/translation#integrating-translations)
- [ ] Update any relevant READMEs and [docs](https://github.com/OriginProtocol/origin/tree/master/docs)

### CODE REVIEW

File: mobile/src/screens/marketplace.js [hunk 1]
```diff
@@ -550,6 +551,13 @@ class MarketplaceScreen extends Component {
     updateExchangeRate(this.state.fiatCurrency[1], 'DAI')
   }
 
+  onWebViewNavigationStateChange = state => {
+    const urlMatch = state.url.match(/^https?:\/\/([^/?#]+)(?:[/?#]|$)/i)
```
Comment (shahthepro) [inline mobile/src/screens/marketplace.js]: nit: We could probably use [URL Web API](https://developer.mozilla.org/en-US/docs/Web/API/URL).

Something like `new URL(state.url).hostname`.

Comment (sparrowDom) [reply mobile/src/screens/marketplace.js]: A lot cleaner your way thanks!

File: mobile/src/screens/marketplace.js [hunk 2]
```diff
@@ -550,6 +551,10 @@ class MarketplaceScreen extends Component {
     updateExchangeRate(this.state.fiatCurrency[1], 'DAI')
   }
 
+  onWebViewNavigationStateChange = state => {
+    this.setState({ currentDomain: new URL(state.url).hostname })
```
Comment (shahthepro) [inline mobile/src/screens/marketplace.js]: It is good to wrap this in `try { ... } catch { ... }` block, unless we are sure that `state.url` will always be a valid URL. `URL` will throw if `state.url` is a malformed URL

Comment (sparrowDom) [reply mobile/src/screens/marketplace.js]: This state is reported from the browser inside webView, so it is probably always going to be a valid URL...

Adding try catch to be extra sure. Defensive programming FTW!

Comment (shahthepro) [reply mobile/src/screens/marketplace.js]: Thanks @sparrowDom. Everything else looks good to me. But Tom may want to take a pass.

Comment (sparrowDom) [reply mobile/src/screens/marketplace.js]: Yup I am sure he will thanks @shahthepro

**Comment by sparrowDom** [comment]
@tomlinton I didn't want to bump the Gradle build versions. This is a P1 bug so we might want to publish the app relatively soon. I can do it, I just might need the google console credentials I think

**Comment by tomlinton** [comment]
Any idea why it didn't like the user agent that function returned? The function was written to try and return an approximation of a standard browser user agent. If twitter doesn't like it it probably means there is some issue there.

The next Android release is going to require quite a bit of testing due to some changes (and 64 bit support for the first time). I'll put a test build up over the weekend and let you know, it'd be great if you have physical devices to test it on.

**Comment by sparrowDom** [comment]
aha big release then nice @tomlinton 

I can't find any official Twitter documentation for this. I think it is a defence mechanism on Twitter's side when they detect something that is not to their liking the running environment. I think it is to prevent possible password password stealing or tweet content manipulation without user knowing.

**Pull request merged**